src/models/dirwatcher.ts

```diff
@@ -0,0 +1,77 @@
+import EventEmitter from "events";
+import fs  from "fs";
+import _ from "lodash";
+import { promisify } from "util";
+
+const readDir = promisify(fs.readdir);
+const stat = promisify(fs.stat);
+
+enum FileStatus {
+  ADDED = "added",
+  MODIFIED = "modified",
+  DELETED = "deleted",
+ }
+
+export class DirWatcher extends EventEmitter {
+  timerId: any;
+  files: string[];
+  fileTempStatistics: object;
+  fileStatistics: object;
+
+  constructor() {
+    super();
+    this.timerId = undefined;
+    this.files = [];
+    this.fileStatistics = {};
+    this.fileTempStatistics = {};
+  }
+
+  checkStatus(file: string): object {
+    const fileLastModified = this.fileStatistics[file],
```

Comment:
`const fileLastModified = ...`
`const fileTimestamp = ...`
Separate the declarations